CMakeLists.txt

```diff
@@ -0,0 +1,516 @@
+	builtin/pull.c builtin/push.c builtin/range-diff.c builtin/read-tree.c 
+	builtin/rebase.c builtin/receive-pack.c builtin/reflog.c builtin/remote.c 
+	builtin/remote-ext.c builtin/remote-fd.c builtin/repack.c builtin/replace.c 
+	builtin/rerere.c builtin/reset.c builtin/rev-list.c builtin/rev-parse.c 
+	builtin/revert.c builtin/rm.c builtin/send-pack.c builtin/shortlog.c 
+	builtin/show-branch.c builtin/show-index.c builtin/show-ref.c 
+	builtin/sparse-checkout.c builtin/stash.c builtin/stripspace.c 
+	builtin/submodule--helper.c builtin/symbolic-ref.c builtin/tag.c 
+	builtin/unpack-file.c builtin/unpack-objects.c builtin/update-index.c 
+	builtin/update-ref.c builtin/update-server-info.c builtin/upload-archive.c 
+	builtin/upload-pack.c builtin/var.c builtin/verify-commit.c builtin/verify-pack.c 
+	builtin/verify-tag.c builtin/worktree.c builtin/write-tree.c)
+
+add_executable(git git.c ${git_SOURCES})
+target_link_libraries(git common-main )
+
+
+add_executable(git-credential-store credential-store.c)
+target_link_libraries(git-credential-store common-main)
+
+add_executable(git-daemon daemon.c)
+target_link_libraries(git-daemon common-main)
+
+add_executable(git-fast-import fast-import.c)
+target_link_libraries(git-fast-import common-main)
+
+add_executable(git-http-backend http-backend.c)
+target_link_libraries(git-http-backend common-main)
+
+add_executable(git-sh-i18n--envsubst sh-i18n--envsubst.c)
+target_link_libraries(git-sh-i18n--envsubst common-main)
+
+add_executable(git-shell shell.c)
+target_link_libraries(git-shell common-main)
+
+if(CURL_FOUND)
+	add_library(http_obj OBJECT http.c)
+
+	add_executable(git-imap-send imap-send.c)
+	target_link_libraries(git-imap-send http_obj common-main ${CURL_LIBRARIES})
+
+	add_executable(git-http-fetch http-walker.c http-fetch.c)
+	target_link_libraries(git-http-fetch http_obj common-main ${CURL_LIBRARIES})
+
+	add_executable(git-remote-http http-walker.c remote-curl.c)
+	target_link_libraries(git-remote-http http_obj common-main ${CURL_LIBRARIES} )
+
+	if(EXPAT_FOUND)
+		add_executable(git-http-push http-push.c)
+		target_link_libraries(git-http-push http_obj common-main ${CURL_LIBRARIES} ${EXPAT_LIBRARIES})
+	endif()
+endif()
+
+add_executable(git-remote-testsvn remote-testsvn.c)
+target_link_libraries(git-remote-testsvn common-main vcs-svn)
+
+add_executable(git-credential-cache credential-cache.c)
+target_link_libraries(git-credential-cache common-main)
+
+add_executable(git-credential-cache--daemon credential-cache--daemon.c)
+target_link_libraries(git-credential-cache--daemon common-main)
+
+
+set(git_builtin_extra
+	cherry cherry-pick format-patch fsck-objects
+	init merge-subtree restore show
+	stage status switch whatchanged)
+
+#Creating hardlinks
+foreach(s ${git_SOURCES} ${git_builtin_extra})
+	string(REPLACE "builtin/" "" s ${s})
+	string(REPLACE ".c" "" s ${s})
+	file(APPEND ${CMAKE_BINARY_DIR}/CreateLinks.cmake "file(CREATE_LINK git git-${s})\n")
```

Comment:
Is this `CreateLinks.cmake` the reason for the minimum CMake version of 3.14? If so, I would like to see that explanation in the commit message.

Speaking of the commit message: it is a bit short. I would like to see paragraphs discussing at least the following topics:
- _why_ is it a good idea to support CMake? (cross-platform, in particular Windows, describing the current `make vcxproj` situation as non-ideal because it requires a full Git for Windows SDK, which weighs in with ~2GB on disk)
- _how_ can this be used?
- in contrast to earlier attempts, why is it now easier to maintain this CMake support? (we make extensive use of CI/PR builds, and this patch series exercises the CMake support in those builds)
- this is only the first patch, adding support for configuring a Linux build of Git via CMake. Subsequent patches will enhance this, step by step, in the interest of making it easier to review.

BTW I could imagine that we'll split out the cURL parts into yet another add-on patch, and maybe even the iconv/intl stuff, to make it _even_ easier to review (a 516-line `CMakeLists.txt` file will probably still scare core Git reviewers, in particular because most of those reviewers strike me as unfamiliar with CMake, I know that _I_ am unfamiliar with that system despite having to use it in a couple of Git for Windows' components).